src/coverlet.core/Abstracts/ILogger.cs

```diff
@@ -1,9 +1,11 @@
 using System;
+using coverlet.core.Enums;
 
 namespace Coverlet.Core.Abstracts
 {
     internal interface ILogger
     {
+        LogLevel Level { get; set; }
```

Comment:
You don't have to move this piece of code...ILogger interface is stable...the "level" is needed only for console driver so the implementation remains there with level and you build the container inside `app.OnExecute` so you can register proper console logger level.

Reply:
Sorry I don't quite understand it. In ```Coverlet.Console``` when I build the container in ```app.OnExecute``` there is no logger when ```app.OnExecute``` throws before ```ILogger``` is resolved. 
That's why I created the container directly in main before ```app.OnExecute``` is invoked. That's also the reason why I added the ```Level``` property to the interface and moved the enum. An alternative would be a direct cast of the resolved ```ILogger``` to ```ConsoleLogger``` but I guess this would break the benefit of using a DI container.

Reply:
You're right, for now register a singleton and cast to logger inside `OnExecute`, I need to understand if we can use IOptions<T> pattern in some way to register an option object with default at the start and update inside `OnExecute` and for every new instance the options is used.